builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -275,9 +277,14 @@ func resourceAwsS3BucketObjectRead(d *schema.ResourceData, meta interface{}) err
 			Key:    aws.String(key),
 		})
 	if err != nil {
-		return fmt.Errorf("Failed to get object tags (bucket: %s, key: %s): %s", bucket, key, err)
+		// Treat the inability to get object tags in a restricted regions as a
+		// soft error.
+		if !restricted {
+			return fmt.Errorf("Failed to get object tags (bucket: %s, key: %s): %s", bucket, key, err)
```

Comment:
I think that's what @apparentlymart was suggesting - also tweaking put to warn users off setting tags.

The benefit to this approach is that if/when these partitions do get support for tagging, it will "fail open" and allow them to be set.